diff --git a/package.json b/package.json
index d38e6dfd..54ebee37 100644
--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
     "defaults"
   ],
   "engines": {
-    "node": ">=6.9",
+    "node": ">=20.0.0",
     "npm": ">=3.0.0"
   },
   "author": "Roman Hotsiy <gotsijroman@gmail.com>",
diff --git a/tsconfig.json b/tsconfig.json
index 7c4df771..582c041f 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,9 +1,10 @@
 {
   "compilerOptions": {
     "experimentalDecorators": true,
-    "moduleResolution": "node",
     "target": "es5",
-    "noImplicitAny": false,
+    "module": "ESNext",
+    "moduleResolution": "Bundler",
+    "resolveJsonModule": true,
     "noUnusedParameters": true,
     "noUnusedLocals": true,
     "strictNullChecks": true,
diff --git a/webpack.config.ts b/webpack.config.ts
index 5268718c..4f77e5cd 100644
--- a/webpack.config.ts
+++ b/webpack.config.ts
@@ -1,4 +1,5 @@
 /* tslint:disable:no-implicit-dependencies */
+import { readFileSync } from 'fs';
 import ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 import * as webpack from 'webpack';
 import * as path from 'path';
@@ -34,9 +35,8 @@ const BANNER = `ReDoc - OpenAPI/Swagger-generated API Reference Documentation
   Repo: https://github.com/Redocly/redoc`;
 
 export default (env: { standalone?: boolean; browser?: boolean } = {}) => ({
-  entry: env.standalone ? ['./src/polyfills.ts', './src/standalone.tsx'] : './src/index.ts',
+  entry: env.standalone ? ['react-dom/client', './src/polyfills.ts', './src/standalone.tsx'] : './src/index.ts',
   output: {
-    filename: env.standalone
       ? 'redoc.standalone.js'
       : env.browser
       ? 'redoc.browser.lib.js'
@@ -85,7 +85,7 @@ export default (env: { standalone?: boolean; browser?: boolean } = {}) => ({
         loader: 'esbuild-loader',
         options: {
           target: 'es2015',
-          tsconfigRaw: require('./tsconfig.json'),
+          tsconfigRaw: JSON.parse(readFileSync(path.resolve(__dirname, 'tsconfig.json'), 'utf-8')),
         },
         exclude: [/node_modules/],
       },
