diff --git a/cypress.config.ts b/cypress.config.ts
index cd67b971..e353b375 100644
--- a/cypress.config.ts
+++ b/cypress.config.ts
@@ -17,4 +17,15 @@ export default defineConfig({
     specPattern: 'e2e/integration/**/*.{js,jsx,ts,tsx}',
     supportFile: false,
   },
+
+  component: {
+    devServer: {
+      framework: 'react',
+      bundler: 'webpack',
+      webpackConfig: require('./webpack.config.ts'),
+    },
+    specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
+    supportFile: 'cypress/support/component.ts',
+    indexHtmlFile: 'cypress/support/component-index.html',
+  },
 });
diff --git a/package.json b/package.json
index d38e6dfd..a03c0003 100644
--- a/package.json
+++ b/package.json
@@ -35,11 +35,12 @@
     "test": "npm run unit && npm run license-check",
     "unit": "jest --coverage",
     "test:update-snapshot": "jest --updateSnapshot",
-    "e2e": "cypress run",
-    "e2e-ci": "cypress run --record",
     "bundlesize": "size-limit",
     "ts-check": "tsc --noEmit --skipLibCheck",
+    "test:types": "tsc --noEmit",
     "cy:open": "cypress open",
+    "cy:component": "cypress run --component",
+    "test:parallel": "npm run test:types & npm run cy:component",
     "bundle:clean": "rimraf bundles",
     "bundle:standalone": "webpack --env production --env standalone --mode=production",
     "bundle:lib": "webpack --mode=production && npm run declarations",
diff --git a/tsconfig.json b/tsconfig.json
index 7c4df771..640ff8f5 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -14,8 +14,7 @@
     "outDir": "lib",
     "pretty": true,
     "lib": ["es2015", "es2016", "es2017", "dom", "WebWorker.ImportScripts"],
-    "jsx": "react",
-    "types": ["webpack", "webpack-env", "jest"]
+    "types": ["webpack", "webpack-env", "jest", "cypress-ct"]
   },
   "compileOnSave": false,
   "exclude": ["node_modules", ".tmp", "lib", "e2e/**"],
diff --git a/webpack.config.ts b/webpack.config.ts
index 5268718c..a92d006b 100644
--- a/webpack.config.ts
+++ b/webpack.config.ts
@@ -1,11 +1,11 @@
 /* tslint:disable:no-implicit-dependencies */
+import * as path from 'path';
 import ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 import * as webpack from 'webpack';
-import * as path from 'path';
+import ReactRefreshWebpackPlugin = require('@pmmmwh/react-refresh-webpack-plugin');
 import { webpackIgnore } from './config/webpack-utils';
 
 const nodeExternals = require('webpack-node-externals')({
-  // bundle in modules that need transpiling + non-js (e.g. css)
   allowlist: [
     'swagger2openapi',
     'marked',
@@ -16,7 +16,6 @@ const nodeExternals = require('webpack-node-externals')({
     /\.(?!(?:jsx?|json)$).{1,5}$/i,
   ],
 });
-
 const VERSION = JSON.stringify(require('./package.json').version);
 let REVISION;
 
@@ -104,11 +103,18 @@ export default (env: { standalone?: boolean; browser?: boolean } = {}) => ({
       },
     ],
   },
+  devServer: {
+    hot: true,
+    client: {
+      progress: true,
+      overlay: true,
+    },
+  },
   plugins: [
+    new webpack.HotModuleReplacementPlugin(),
+    new ReactRefreshWebpackPlugin(),
     new webpack.DefinePlugin({
-      __REDOC_VERSION__: VERSION,
       __REDOC_REVISION__: REVISION,
-      'process.env': '{}',
       'process.platform': '"browser"',
       'process.stdout': 'null',
     }),
